mlflow/pyfunc/model.py

```diff
@@ -0,0 +1,275 @@
+                      Conda environment yaml file. If provided, this decribes the environment
+                      this model should be run in. At minimum, it should specify the dependencies
+                      contained in :data:`mlflow.pyfunc.DEFAULT_CONDA_ENV`. If `None`, the default
+                      :data:`mlflow.pyfunc.DEFAULT_CONDA_ENV` environment will be added to the
+                      model.
+    :return: Model configuration containing model info.
+    """
+    if loader_module is None:
+        raise MlflowException(
+            message=("`loader_module` must be specified!"),
+            error_code=INVALID_PARAMETER_VALUE)
+
+    if os.path.exists(path):
+        raise MlflowException(
+                message="Path '{}' already exists".format(path),
+                error_code=RESOURCE_ALREADY_EXISTS)
+    os.makedirs(path)
+
+    code = None
+    data = None
+    env = None
+
+    if data_path is not None:
+        model_file = _copy_file_or_tree(src=data_path, dst=path, dst_dir="data")
+        data = model_file
+
+    if code_paths is not None:
+        for code_path in code_paths:
+            _copy_file_or_tree(src=code_path, dst=path, dst_dir="code")
+        code = "code"
+
+    if conda_env is not None:
+        shutil.copy(src=conda_env, dst=os.path.join(path, "mlflow_env.yml"))
+        env = "mlflow_env.yml"
+
+    mlflow.pyfunc.add_to_model(
+        mlflow_model, loader_module=loader_module, code=code, data=data, env=env)
+    mlflow_model.save(os.path.join(path, 'MLmodel'))
+    return mlflow_model
+
+
+def _load_pyfunc(model_path):
+    pyfunc_config = _get_flavor_configuration(
+            model_path=model_path, flavor_name=mlflow.pyfunc.FLAVOR_NAME)
+    python_model_subpath = pyfunc_config.get(CONFIG_KEY_PYTHON_MODEL, None)
+    if python_model_subpath is None:
+        raise MlflowException(
+            "Python model path was not specified in the model configuration")
+    with open(os.path.join(model_path, python_model_subpath), "rb") as f:
+        python_model = cloudpickle.load(f)
+
+    directory_managers = []
+    if sys.version_info >= (3, 2):
+        # Create a managed temporary directory that will exist as long as the manager object
+        # returned by `tempfile.TemporaryDirectory` is in scope. This directory will be removed
+        # some time after the manager object goes out of scope.
+        tmp_artifacts_dir_manager = tempfile.TemporaryDirectory(suffix="artifacts")
+        directory_managers.append(tmp_artifacts_dir_manager)
+        tmp_artifacts_dir_path = tmp_artifacts_dir_manager.name
+    else:
+        # Because `tempfile.TemporaryDirectory` does not exist prior to Python 3.2, create an
+        # unmanaged temporary directory instead. Depending on the system, this directory is likely
+        # to be created in "/var" or "/tmp" and will be removed on system reboot.
+        # TODO: If the longevity of the temporary directory prior to Python 3.2 becomes problematic,
```

Comment:
This is probably a premature optimization in general; would rather add this directory_managers feature later.